plugins/woocommerce-blocks/assets/js/blocks/cart/inner-blocks/cart-order-summary-subtotal/edit.tsx

```diff
@@ -1,26 +1,54 @@
 /**
  * External dependencies
  */
-import { useBlockProps } from '@wordpress/block-editor';
+import { RichText, useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { useCallback, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import Block from './block';
+import Block, { BlockAttributes } from './block';
+import { DEFAULT_HEADING } from './constants';
 
 export const Edit = ( {
 	attributes,
-}: {
-	attributes: {
-		className: string;
-	};
-	setAttributes: ( attributes: Record< string, unknown > ) => void;
-} ): JSX.Element => {
-	const { className } = attributes;
+	setAttributes,
+}: BlockEditProps< BlockAttributes > ) => {
+	const { className, heading } = attributes;
 	const blockProps = useBlockProps();
+
+	const [ headingText, setHeadingText ] = useState(
+		heading || DEFAULT_HEADING
+	);
+
+	const onChangeCallback = useCallback(
+		( value: string ) => {
+			setHeadingText( value );
```

Comment:
For the editor we maintain a piece of state to show the rich text edited value, this is so that we can display the up to date edited value to the user, even if maybe we reset the attribute under the hood, such as we do [here](https://github.com/woocommerce/woocommerce/pull/53544/files#diff-090b2440beb63e4a22de7aac648a50ee7cb9cc3023d39e4c9033cca4dab09e60R30)

Reply:
Nice way of ensuring that the translation remains working when a merchant resets the heading. Well done, @samueljseay. 👏